packages/flutter_tools/lib/src/base/platform.dart

```diff
@@ -176,89 +174,119 @@ class LocalPlatform extends Platform {
 /// Provides a mutable implementation of the [Platform] interface.
 class FakePlatform extends Platform {
   /// Creates a new [FakePlatform] with the specified properties.
-  ///
-  /// Unspecified properties will *not* be assigned default values (they will
-  /// remain `null`).
   FakePlatform({
-    this.numberOfProcessors,
-    this.pathSeparator,
-    this.operatingSystem,
-    this.operatingSystemVersion,
-    this.localHostname,
-    this.environment,
-    this.executable,
-    this.resolvedExecutable,
-    this.script,
-    this.executableArguments,
+    this.numberOfProcessors = 1,
+    String? pathSeparator,
+    required this.operatingSystem,
+    String? operatingSystemVersion,
```

Comment:
Does it make more sense to add `operatingSystemVersion = 'linux'` here instead of all the test call locations?  Usually the os doesn't matter for tests, until it does and you want to be explicit anyway.